driver-core/src/main/com/mongodb/UnixServerAddress.java

```diff
@@ -44,14 +44,18 @@ public UnixServerAddress(final String path) {
         isTrueArgument("The path must end in .sock", path.endsWith(".sock"));
     }
 
+    @SuppressWarnings("deprecation")
```

Comment:
I'm not sure why the warning also has to be suppressed here, but it complains if I don't

Reply:
It's probably because of the [`-Xlint:deprecation`](https://docs.oracle.com/en/java/javase/17/docs/specs/man/javac.html#option-deprecation) lint: the method overrides a deprecated method.